lib/ansible/modules/cloud/azure/azure_rm_trafficmanager.py

```diff
@@ -0,0 +1,580 @@
+                required: true
+            target_resource_id:
+                description:
+                    - The Azure Resource URI of the of the endpoint.
+                    - Not applicable to endpoints of type 'ExternalEndpoints'.
+            target:
+                description:
+                    - The fully-qualified DNS name of the endpoint.
+            endpoint_status:
+                description:
+                    - The status of the endpoint.
+                choices:
+                    - Enabled
+                    - Disabled
+            weight:
+                description:
+                    - The weight of this endpoint when using the 'Weighted' traffic routing method.
+                    - Possible values are from 1 to 1000.
+            priority:
+                description:
+                    - The priority of this endpoint when using the 'Priority' traffic routing method.
+                    - Possible values are from 1 to 1000, lower values represent higher priority.
+                    - This is an optional parameter. If specified, it must be specified on all endpoints.
+                    - No two endpoints can share the same priority value.
+            endpoint_location:
+                description:
+                    - Specifies the location of the external or nested endpoints when using the 'Performance' traffic routing method.
+            min_child_endpoints:
+                description:
+                    - The minimum number of endpoints that must be available in the child profile in order for the parent profile to be considered available.
+                    - Only applicable to endpoint of type 'NestedEndpoints'.
+            geo_mapping:
+                description:
+                    - The list of countries/regions mapped to this endpoint when using the 'Geographic' traffic routing method.
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Hai Cao <t-haicao@microsoft.com>"
+
+'''
+
+EXAMPLES = '''
+    - name: Create a Traffic Manager Profile
+      azure_rm_trafficmanager:
+        name: tmtest
+        resource_group: tmt
+        location: global
+        profile_status: Enabled
+        traffic_routing_method: Priority
+        dns_config:
+          relative_name: tmtest
+          ttl: 60
+        monitor_config:
+          protocol: HTTPS
+          port: 80
+          path: '/'
+        endpoints:
+          - name: e1
+            type: Microsoft.network/TrafficManagerProfiles/ExternalEndpoints
+            endpoint_location: West US 2
+            endpoint_status: Enabled
+            priority: 2
+            target: 1.2.3.4
+            weight: 1
+        tags:
+          Environment: Test
+
+    - name: Delete a Traffic Manager Profile
+      azure_rm_trafficmanager:
+        state: absent
+        name: tmtest
+        resource_group: tmt
+'''
+RETURN = '''
+state:
```

Comment:
Just return the `id` of the traffic manager profile (and maybe a list of endpoint ids). We should only be returning things the user may not know. We use the _facts module to return all the remaining information in an easier format. So this should be

```
RETURN = '''
id:
  description: The ID of the traffic manager profile
  returned: always
  type: str
  example: "/subscriptions/XXXXXX...XXXXXXXXX/resourceGroups/tmt/providers/Microsoft.Network/trafficManagerProfiles/tmtest"
endpoints:
  description: List of endpoint IDs attached to the profile
  returned: always
  type: list
  sample: [ "id1...", "id2..." ]
'''
```

Reply:
fixed